strax/processing/statistics.py

```diff
@@ -42,89 +28,96 @@ def highest_density_region(data, fractions_desired, only_upper_part=False, _buff
         )
 
     # Need an index which sorted by amplitude
-    max_to_min = np.argsort(data, kind="mergesort")[::-1]
+    max_to_min = stable_argsort(data)[::-1]
+    return max_to_min, area_tot, res, res_amp, fi
+
+
+@export
+def _process_hdr_intervals(ind, gaps, fi, res, g0, _buffer_size):
+    """Process the intervals for highest density region.
+
+    This function handles the stable sorting part outside of numba.
+
+    """
+    if len(gaps) > _buffer_size:
+        res[fi, 0, :] = -1
+        res[fi, 1, :] = -1
+        return fi + 1, res
+
+    g_ind = -1
+    for g_ind, g in enumerate(gaps):
+        interval = ind[g0:g]
+        res[fi, 0, g_ind] = interval[0]
+        res[fi, 1, g_ind] = interval[-1] + 1
+        g0 = g
+
+    # Last interval
+    interval = ind[g0:]
+    res[fi, 0, g_ind + 1] = interval[0]
+    res[fi, 1, g_ind + 1] = interval[-1] + 1
+    return fi + 1, res
+
+
+@export
+@register_jitable
+def highest_density_region(data, fractions_desired, only_upper_part=False, _buffer_size=10):
```

Comment:
Are we sure that the behavior of `highest_density_region` is not changed? It is used at https://github.com/XENONnT/cutax/blob/195b6922cf5dc27da23fb2896a252f557be80752/cutax/cut_efficiency.py#L194.

Reply:
Originally it was using the `numba` implemented `sort`, which uses `quicksort`. But now we avoid using the `numba` one, only call the pure Python function, and enforce `mergesort`. So I would suppose the behavior changes.